File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.py
class Solution:
    def invert(self, s: str) -> str:
        inverted_str = ""
        for char in s:
            if char == '0':
                inverted_str += '1'
            else:
                inverted_str += '0'
        return inverted_str

    def findKthBit(self, n: int, k: int) -> str:
        s = [""] * n
        s[0] = "0"

        for i in range(1, n):
            reversed_str = self.invert(s[i - 1])[::-1]  # Invert and reverse the string
            s[i] = s[i - 1] + "1" + reversed_str

        if 0 <= k - 1 < len(s[n - 1]):
            return s[n - 1][k - 1]

File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.cpp
class Solution {
public:
    string invert(string s) {
        string inverted_str = "";
        for(int i = 0; i < s.length(); i++) {
            if(s[i] == '0') {
                inverted_str += '1';
            } else {
                inverted_str += '0';
            }
        }
        return inverted_str;
    }
    char findKthBit(int n, int k) {
        vector<string> s(n,"");
        char kth_bit;
        s[0] = "0";
        for(int i = 1; i < n; i++) {
            string reversed_str = invert(s[i - 1]);
            reverse(reversed_str.begin(), reversed_str.end());
            s[i] = s[i - 1] + "1" + reversed_str;
        }
        for(int i = 0; i < s[n-1].length(); i++){
            if(i == k-1){
                kth_bit = s[n-1][i];
            }
        } 
        return kth_bit;
    }
};